### PR TITLE
feat(react-query): add `<SuspenseQuery/>`, `<SuspenseInfiniteQuery/>`

### LINKED ISSUE #771: [Feature]: v2 - SuspenseQuery, SuspenseQueries, SuspenseInfinteQuery to wrap inline in jsx easily
### Package Scope

@suspensive/react-query

### Description

```tsx
'use client'

import { Suspense } from '@suspensive/react'
import {
  type QueryKey,
  type UseSuspenseQueryOptions,
  type UseSuspenseQueryResult,
  queryOptions,
  useSuspenseQuery,
} from '@tanstack/react-query'
import { type ReactNode, useState } from 'react'

// SuspenseQuery.ts
export const SuspenseQuery = < // same with interface of useSuspenseQuery
  TQueryFnData = unknown,
  TError = Error,
  TData = TQueryFnData,
  TQueryKey extends QueryKey = QueryKey,
>({
  children,
  ...options
}: UseSuspenseQueryOptions<TQueryFnData, TError, TData, TQueryKey> & {
  children: (result: UseSuspenseQueryResult<TData, TError>) => ReactNode
}) => children(useSuspenseQuery(options))

// api.ts
const api = {
  users: {
    getList: (filter: { limit: number }) => Promise.resolve('success' + JSON.stringify(filter)),
  },
}

// query.ts
const query = {
  users: {
    getList: (...params: Parameters<typeof api.users.getList>) =>
      queryOptions({
        queryKey: ['query.users.getList', params] as const,
        queryFn: () => Promise.resolve([{ id: 0 }, { id: 1 }]),
      }),
  },
}

// UserListPage.tsx
const UserListPage = () => {
  const [selectedId, setSelectedId] = useState<number | null>(null)

  return (
    <Suspense fallback="user loading...">
      <SuspenseQuery {...query.users.getList({ limit: 5 })}>
        {({ data: userList }) => userList.map((user) => (
          <ul>
            <li key={user.id}>
              {user.id}
              <button onClick={() => setSelectedId(user.id)}>select user</button>
            </li>
          </ul>
        )}
      </SuspenseQuery>
      {selectedId !== null ? (
        <SuspenseQuery
          {...query.users.getList({ limit: 5 })}
          select={(userList) => userList.find(({ id }) => id === selectedId)} // re-use query.users.getList with select
        >
          {({ data: userList }) => userList.map((user) => <>{user.name</>)}
        </SuspenseQuery>)
      : null}
      <Suspense fallback="selected user loading...">
        {selectedId !== null ? (
          <SuspenseQuery
            {...query.users.info({ userId: selectedId })}
          >
            {({ data: user}) => <>{user.description}</>}
          </SuspenseQuery>
        ) : null}
      </Suspense>
    </Suspense>
  )
}
```

### Possible Solution

_No response_

### etc.

_No response_

### PR DESCRIPTION
**Pull request by manudeli**
close #771 

## PR Checklist

- [x] I did below actions if need

1. I read the [Contributing Guide](https://github.com/suspensive/react/blob/main/CONTRIBUTING.md)
2. I added documents and tests.

### CODE REVIEW

File: .changeset/rich-flies-turn.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@suspensive/react-query": patch
```
Comment (manudeli) [inline .changeset/rich-flies-turn.md:2]:
```suggestion
"@suspensive/react-query": major
```

File: packages/react-query/src/useSuspenseInfiniteQuery.ts [hunk 2]
```diff
@@ -31,50 +29,16 @@ export interface UseSuspenseInfiniteQueryOptions<
...
-export function useSuspenseInfiniteQuery<
-  TQueryFnData = unknown,
-  TError = unknown,
-  TData = TQueryFnData,
-  TQueryKey extends QueryKey = QueryKey,
->(
-  queryKey: TQueryKey,
-  options: Omit<UseSuspenseInfiniteQueryOptions<TQueryFnData, TError, TData, TQueryKey>, 'queryKey'>
-): UseSuspenseInfiniteQueryResult<TData, TError>
-// arg1: options
```
Comment (manudeli) [inline packages/react-query/src/useSuspenseInfiniteQuery.ts:55]: From @suspensive/react-query v2, there is only one parameter to pass infiniteQueryOptions

File: packages/react-query/src/useSuspenseQuery.ts [hunk 3]
```diff
@@ -27,48 +20,14 @@ export interface UseSuspenseQueryOptions<
...
-  TError = unknown,
-  TData = TQueryFnData,
-  TQueryKey extends QueryKey = QueryKey,
->(
-  arg1: TQueryKey | UseSuspenseQueryOptions<TQueryFnData, TError, TData, TQueryKey>,
-  arg2?:
-    | QueryFunction<TQueryFnData, TQueryKey>
-    | Omit<UseSuspenseQueryOptions<TQueryFnData, TError, TData, TQueryKey>, 'queryKey'>,
-  arg3?: Omit<UseSuspenseQueryOptions<TQueryFnData, TError, TData, TQueryKey>, 'queryKey' | 'queryFn'>
-) {
```
Comment (manudeli) [inline packages/react-query/src/useSuspenseQuery.ts:69]: From @suspensive/react-query v2, there is only one parameter to pass queryOptions

File: packages/react-query/src/useSuspenseQuery.ts [hunk 4]
```diff
@@ -27,48 +20,14 @@ export interface UseSuspenseQueryOptions<
...
+>(options: UseSuspenseQueryOptions<TQueryFnData, TError, TData, TQueryKey>) {
   return useQuery<TQueryFnData, TError, TData, TQueryKey>({
-    ...parseQueryArgs(arg1, arg2, arg3),
```
Comment (manudeli) [inline packages/react-query/src/useSuspenseQuery.ts:71]: there is no need to use parseQueryArgs for aggregate arg1, arg2, arg3 from now

File: packages/react-query/src/useSuspenseInfiniteQuery.ts [hunk 5]
```diff
@@ -31,50 +29,16 @@ export interface UseSuspenseInfiniteQueryOptions<
...
-  TQueryKey extends QueryKey = QueryKey,
->(
-  arg1: TQueryKey | UseSuspenseInfiniteQueryOptions<TQueryFnData, TError, TData, TQueryKey>,
-  arg2?:
-    | QueryFunction<TQueryFnData, TQueryKey>
-    | Omit<UseSuspenseInfiniteQueryOptions<TQueryFnData, TError, TData, TQueryKey>, 'queryKey'>,
-  arg3?: Omit<UseSuspenseInfiniteQueryOptions<TQueryFnData, TError, TData, TQueryKey>, 'queryKey' | 'queryFn'>
 ): UseSuspenseInfiniteQueryResult<TData, TError> {
   return useInfiniteQuery({
-    ...parseQueryArgs(arg1, arg2, arg3),
```
Comment (manudeli) [inline packages/react-query/src/useSuspenseInfiniteQuery.ts:77]: there is no need to use parseQueryArgs for aggregate arg1, arg2, arg3 from now

**Review by manudeli — COMMENTED** [review]
comments as intention

**Comment by tooooo1** [comment]
shoutout [with-function crew](https://github.com/function-and-mountain) 🔥

**Pull request merged**